src/abstractdatatable/abstractdatatable.jl

```diff
@@ -801,6 +738,214 @@ function Base.hash(dt::AbstractDataTable)
     return @compat UInt(h)
 end
 
+"""
+    denullify!(dt::AbstractDataTable)
+
+Convert `NullableArray` & `Array{Nullable}` columns that do not contain null values
+to a non-`Nullable` equivalent array type. The table `dt` is modified in place.
+
+# Examples
+
+```jldoctest
+julia> dt = DataTable(A = NullableArray(1:3), B = [Nullable(i) for i=1:3])
+3×2 DataTables.DataTable
+│ Row │ A │ B │
+├─────┼───┼───┤
+│ 1   │ 1 │ 1 │
+│ 2   │ 2 │ 2 │
+│ 3   │ 3 │ 3 │
+
+julia> eltypes(dt)
+2-element Array{Type,1}:
+ Nullable{Int64}
+ Nullable{Int64}
+
+julia> eltypes(denullify!(dt))
+2-element Array{Type,1}:
+ Int64
+ Int64
+
+julia> eltypes(dt)
+2-element Array{Type,1}:
+ Int64
+ Int64
+```
+
+See also [`denullify`](@ref).
+"""
+function denullify!(dt::AbstractDataTable)
+    for i in 1:size(dt,2)
+        dt[i] = denullify(dt[i])
+    end
+    dt
+end
+
+"""
+    denullify(dt::AbstractDataTable)
+
+Convert `NullableArray` & `Array{Nullable}` columns that do not contain null values
+to a non-`Nullable` equivalent array type.
+
+# Examples
+
+```jldoctest
+julia> dt = DataTable(A = NullableArray(1:3), B = [Nullable(i) for i=1:3])
+3×2 DataTables.DataTable
+│ Row │ A │ B │
+├─────┼───┼───┤
+│ 1   │ 1 │ 1 │
+│ 2   │ 2 │ 2 │
+│ 3   │ 3 │ 3 │
+
+julia> eltypes(dt)
+2-element Array{Type,1}:
+ Nullable{Int64}
+ Nullable{Int64}
+
+julia> eltypes(denullify(dt))
+2-element Array{Type,1}:
+ Int64
+ Int64
+
+julia> eltypes(dt)
+2-element Array{Type,1}:
+ Nullable{Int64}
+ Nullable{Int64}
+```
+
+See also [`denullify!`](@ref).
+"""
+denullify(dt::AbstractDataTable) = denullify!(copy(dt))
+
+
+"""
+    denullify(A::AbstractArray)
+
+Convert `NullableArray` & `Array{Nullable}` arrays that do not contain null values
+to a non-`Nullable` equivalent array type.
+
+# Examples
+
+```jldoctest
+julia> A = NullableArray(1:3)
+3-element NullableArrays.NullableArray{Int64,1}:
+ 1
+ 2
+ 3
+
+julia> denullify(A)
+3-element Array{Int64,1}:
+ 1
+ 2
+ 3
+
+julia> B = [Nullable(i) for i=1:3]
+3-element Array{Nullable{Int64},1}:
+ 1
+ 2
+ 3
+
+julia> denullify(B)
+3-element Array{Int64,1}:
+ 1
+ 2
+ 3
+```
+
+"""
+function denullify(A::AbstractVector)
+    if (eltype(A) <: Nullable) && !any(x -> isnull(x), A)
+        if isa(A, NullableArray)
```

Comment:
Also needs a special case for `NullableCategoricalArray`. Actually I wonder whether this couldn't just call `dropnull` to avoid duplicating the logic. If a new custom type wants to work with `DataTable`, the current approach won't allow it to override the method. It would be better if it could just override `dropnull`.